ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -484,27 +486,28 @@ encodeShelleyLedgerState
                        , shelleyLedgerTransition
                        } =
     encodeVersion serialisationFormatVersion2 $ mconcat [
-        CBOR.encodeListLen 2
+        CBOR.encodeListLen 3
       , encodeWithOrigin encodeShelleyTip shelleyLedgerTip
       , toCBOR shelleyLedgerState
       , encodeShelleyTransition shelleyLedgerTransition
       ]
 
 decodeShelleyLedgerState ::
      forall era s. ShelleyBasedEra era
-  => Decoder s (LedgerState (ShelleyBlock era))
-decodeShelleyLedgerState = decodeVersion [
+  => Decoder s (Lazy.ByteString -> LedgerState (ShelleyBlock era))
```

Comment:
I am now considering if it would be better to do the cbor-in-cbor encoding in the ledger package itself, so that there could be a `FromCBOR` instance of `LedgerState` and thus break less downstream. I'll maybe tackle that on Monday if the others agree.

Reply:
@JaredCorduan Edsko and I don't have a strong opinion on where it belongs, though it does seem more like a ledger implementation detail, especially based on how small this PR's diff is.

Reply:
I've talked with @nc6 about this, and we decided that not having these odd cbor-in-cbor `FromCBOR` instances is probably best. I think that storing ledger state to disk using the cbor-in-cbor encoding is less disruptive than other options (and I would probably need help with the other options, namely percolating the annotators up through the types), but others may have a different opinion. I'm not sure he consequences of using the cbor-in-cbor encoding in the queries.